test/verify/check-s4u-ssh

```diff
@@ -0,0 +1,95 @@
+        # add user to impersonate
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa user-add --first=user --last=user user'")
+        # ipa_machine.execute("echo foobarfoo | ipa user-mod user --password")
+        # add service which will be allowed to delegate creds
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-add cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN --ok-as-delegate=true --ok-to-auth-as-delegate=true'")
+        # Allow restrieval of service/host keytabs by admin
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa service-allow-retrieve-keytab --user=admin cockpitclient/sshclient.cockpit.lan'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa host-allow-retrieve-keytab --user=admin sshserver.cockpit.lan'")
+        # set up delegation rule
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationtarget-add cockpit-target'")
+
+        # replace with
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationtarget-add-member cockpit-target --principals=\"host/sshserver.cockpit.lan@COCKPIT.LAN\"'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add cockpit-delegation'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add-member cockpit-delegation --principals=\"cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN\"'")
+        ipa_machine.execute("docker exec freeipa bash -c 'ipa servicedelegationrule-add-target cockpit-delegation --servicedelegationtargets=\"cockpit-target\"'")
+
+        # ssh client, get keytab for authenticating and make ccache using gssapi
+        client_machine.execute("echo foobarfoo | kinit admin@COCKPIT.LAN")
+        client_machine.execute("ipa-getkeytab -k /etc/cockpit-client.keytab -p cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN")
+
+        client_machine.execute("kinit -k -t /etc/cockpit-client.keytab cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN")
+        client_machine.write('/tmp/make-cache.py', """
+import gssapi
+
+principal = gssapi.Name('cockpitclient/sshclient.cockpit.lan@COCKPIT.LAN', gssapi.NameType.kerberos_principal)
+credResult = gssapi.creds.rcred_cred_store.acquire_cred_from(store = { 'keytab'.encode(): '/etc/cockpit-client.keytab'.encode() }, name=principal)
+impersonee = gssapi.Name('user@COCKPIT.LAN', gssapi.NameType.kerberos_principal)
+credResults4u2self = gssapi.creds.rcred_s4u.acquire_cred_impersonate_name(credResult.creds, impersonee)
+gssapi.creds.rcred_cred_store.store_cred_into({ 'ccache'.encode(): 'FILE:/etc/cockpit.ccache'.encode(), 'keytab'.encode(): '/etc/cockpit-client.keytab'.encode() }, credResults4u2self.creds, overwrite=True)
+""")
+        client_machine.execute("python /tmp/make-cache.py")
+
+        # enable gssapiauth
+        sshd_machine.execute("sed -i 's/#GSSAPIAuthentication yes/GSSAPIAuthentication yes/' /etc/ssh/sshd_config")
```

Comment:
Perhaps on Debian/Ubuntu etc. the default file looks differently and doesn't have a commented-out version of this? Also, on latest  Fedora and RHEL, ssh now supports /etc/ssh/ssh_config.d/ , which is rather convenient for cases like this. But I suppose that's not available on older OSes yet :sob:

Reply:
I think the default there is "#GSSAPIAuthentication no" :D